haigha/transports/socket_transport.py

```diff
@@ -29,26 +34,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
-        self._sock.setblocking(True)
-        self._sock.settimeout(self.connection._connect_timeout)
-        if self.connection._sock_opts:
-            for k, v in self.connection._sock_opts.iteritems():
-                family, type = k
-                self._sock.setsockopt(family, type, v)
-        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
-        exc = socket.error("getaddrinfo returns an empty list")
-        for _family, _socktype, _proto, _canonname, sockaddr in infos:
+
+        for info in socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP):
+
+            family, socktype, proto, _, sockaddr = info
+            self._sock = klass(family, socktype, proto)
+            self._sock.setblocking(True)
+            self._sock.settimeout(self.connection._connect_timeout)
+            if self.connection._sock_opts:
+                for k, v in self.connection._sock_opts.iteritems():
+                    family, type = k
```

Comment:
These variable names are incorrect. The definition is `socket.setsockopt(level, optname, value)`, shadowing the underlying C definition (see http://linux.die.net/man/2/setsockopt). More appropriate names would be `level` and `option`. Since we're rewriting this method, it's a good opportunity to clean up this legacy fault, especially when it's this easy.

The names `family` and `type` are also poor choices for the following reason: `family` clobbers the local variable `family` declared above, so that if we wish to refer to that family below (e.g., in logging), we can't, because it's been clobbered; `type` is a bad name, because it's bad practice to override builtins.